extensions/amp-facebook-comments/1.0/component.js

```diff
@@ -0,0 +1,110 @@
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {
+  MessageType,
+  ProxyIframeEmbed,
+} from '../../../src/preact/component/3p-frame';
+import {dashToUnderline} from '../../../src/string';
+import {deserializeMessage} from '../../../src/3p-frame-messaging';
+import {forwardRef} from '../../../src/preact/compat';
+import {tryParseJson} from '../../../src/json';
+import {useCallback, useLayoutEffect, useState} from '../../../src/preact';
+
+/** @const {string} */
+const TYPE = 'facebook';
+const FULL_HEIGHT = '100%';
+const MATCHES_MESSAGING_ORIGIN = () => true;
+const DEFAULT_TITLE = 'Facebook comments';
+
+/**
+ * @param {!FacebookCommentsDef.Props} props
+ * @param {{current: (!FacebookCommentsDef.Api|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function FacebookCommentsWithRef(
+  {
+    colorScheme,
+    href,
+    locale: localeProp,
+    numPosts,
+    onReady,
+    orderBy,
+    requestResize,
+    title = DEFAULT_TITLE,
+    ...rest
+  },
+  ref
+) {
+  const [height, setHeight] = useState(FULL_HEIGHT);
+  const messageHandler = useCallback(
+    (event) => {
+      const data = tryParseJson(event.data) ?? deserializeMessage(event.data);
+      if (data['action'] == 'ready') {
+        if (onReady) {
+          onReady();
+        }
+      }
+      if (data['type'] == MessageType.EMBED_SIZE) {
+        const height = data['height'];
+        if (requestResize) {
+          requestResize(height);
+          setHeight(FULL_HEIGHT);
+        } else {
+          setHeight(height);
+        }
+      }
+    },
+    [requestResize, onReady]
+  );
+
+  const [locale, setLocale] = useState(localeProp);
+  useLayoutEffect(() => {
+    if (localeProp) {
+      setLocale(localeProp);
+      return;
+    }
+    const win = ref?.current?.ownerDocument?.defaultView;
+    if (!win) {
+      return;
+    }
+    setLocale(dashToUnderline(win.navigator.language));
+  }, [localeProp, ref]);
+
+  return (
+    <ProxyIframeEmbed
+      contextOptions={{tagName: 'AMP-FACEBOOK-COMMENTS'}}
+      options={{colorScheme, href, locale, numPosts, orderBy}}
+      ref={ref}
+      title={title}
+      {...rest}
+      /* non-overridable props */
+      // We sandbox all 3P iframes however facebook embeds completely break in
+      // sandbox mode since they need access to document.domain, so we
+      // exclude facebook.
```

Comment:
Is the only thing they need access to is `document.domain`? Could we create a Proxy?

Reply:
@jridgewell Do you have more historical context here? This behavior mimics: https://github.com/ampproject/amphtml/blob/2bddbc2c4982a3761c0c388c1e86d62cdf0c055d/src/3p-frame.js#L158-L161

And comment is copied from test page of 0.1:
https://github.com/ampproject/amphtml/blob/2bddbc2c4982a3761c0c388c1e86d62cdf0c055d/extensions/amp-facebook/0.1/test/test-amp-facebook.js#L72-L74

Reply:
We might try adding sandbox again, and see if it breaks. Maybe Facebook has updated their scripts.